eva/udfs/decorators/decorators.py

```diff
@@ -54,8 +55,47 @@ def forward(input_signatures: List[IOArgument], output_signatures: List[IOArgume
 
     def inner_fn(arg_fn):
         def wrapper(*args):
+
+            # checking the user constraints
+            if check_input:
+                if len(input_signatures) > 0:
```

Comment:
nit: Isn't `input_signatures` a list by default? I think we can cleanup the code by just looping over the input_signatures.

Reply:
removed this.